src/controller.rs

```diff
@@ -399,7 +421,22 @@ fn pod_template(config: &WasmCloudHostConfig, _ctx: Arc<Context>) -> PodTemplate
             ]),
             ..Default::default()
         }),
-    }
+    };
+
+    if let Some(scheduling_options) = &config.spec.scheduling_options {
+        if let Some(pod_overrides) = &scheduling_options.pod_template_additions {
+            let mut overrides = pod_overrides.clone();
+            overrides.service_account_name = Some(service_account);
+            overrides.containers = containers.clone();
+            if let Some(ref mut vols) = overrides.volumes {
```

Comment:
Nit: You own `overrides`, so I don't think you need any `mut` stuff here. This can probably just be:

```rust
if let Some(vols) = overrides.volumes {
    volumes.extend(vols);
}
overrides.volumes = Some(volumes);
```